mitiq/shadows/shadows.py

```diff
@@ -0,0 +1,199 @@
+
+        # calibration circuit is of same qubit number with original circuit
+        zero_circuit = cirq.Circuit()
+        # perform random Pauli measurement one the calibration circuit
+        calibration_measurement_outcomes = random_pauli_measurement(
+            zero_circuit,
+            n_total_measurements=num_total_measurements_calibration,
+            executor=executor,
+            qubits=qubits,
+        )
+    else:
+        calibration_measurement_outcomes = zero_state_shadow_outcomes
+    # get the median of means estimation of Pauli fidelities
+    return get_pauli_fidelities(
+        calibration_measurement_outcomes, k_calibration, locality=locality
+    )
+
+
+def shadow_quantum_processing(
+    circuit: cirq.Circuit,
+    executor: Callable[[cirq.Circuit], MeasurementResult],
+    num_total_measurements_shadow: int,
+    random_seed: Optional[int] = None,
+    qubits: Optional[List[cirq.Qid]] = None,
+) -> Tuple[List[str], List[str]]:
+    r"""
+    This function returns the bit strings and Pauli strings corresponding to
+    the executor measurement outcomes for a given circuit, rotated by unitaries
+    randomly sampled from a fixed unitary ensemble :math:`\mathcal{G}`.
+
+    In the current implementation, the unitaries are sampled from the local
+    Clifford group for :math:`n` qubits, i.e.,
+    :math:`\mathcal{G} = \mathrm{Cl}_2^n`.
+
+    Args:
+        circuit: The circuit to execute.
+        executor: The function to use to do quantum measurement,
+            must be same as executor in `pauli_twirling_calibrate`.
+        num_total_measurements_shadow: Total number of shots for shadow
+            estimation.
+        random_seed: The random seed to use for the shadow measurements.
+        qubits: The qubits to measure.
+
+    Returns:
+        A dictionary containing the bit strings, the Pauli strings
+        `bit_strings`: Circuit qubits computational basis
+        e.g. "01..":math:`:=|0\rangle|1\rangle..`.
+        `pauli_strings`: The local Pauli measurement performed on each
+        qubit. e.g."XY.." means perform local X-basis measurement on the
+        1st qubit, local Y-basis measurement the 2ed qubit in the circuit, etc.
+    """
+    if random_seed is not None:
+        np.random.seed(random_seed)
+    r"""
+    Shadow stage 1: Sample random unitary form
+    :math:`\mathcal{g}\subset \mathrm{U}(2^n)` and perform computational
+    basis measurement. In the current state, we have implemented
+    local Pauli measurement, i.e. :math:`\mathcal{g} = \mathrm{Cl}_2^n`.
+    """
+    # random Pauli measurement on the circuit
+    output = random_pauli_measurement(
+        circuit,
+        n_total_measurements=num_total_measurements_shadow,
+        executor=executor,
+        qubits=qubits,
+    )
+    return output
+
+
+def classical_post_processing(
+    shadow_outcomes: Tuple[List[str], List[str]],
+    rshadows: bool = False,
+    calibration_results: Optional[Dict[str, float]] = None,
```

Comment:
```
rshadows: bool = False,
calibration_results: Optional[Dict[str, float]] = None,
```
Do we need the the `rshadows` argument?  Can we deduce if we want robust or non-robust shadow estimation from `calibration_results` beeing `not None` or `None`.

If you agree, we can simply remove the `rshadows` argument. 
Otherwise, if `rshadows` cannot be removed, I would rename `rshadows`-> `robust` or `use_calibration`.

Reply:
I think it's just a reminder to users whether they'll use the pauli_twilling_calibration. This bool argument has been used as an `if` in the post-processing state.  I prefer to keep this argument. I'll change it to use_calibration.